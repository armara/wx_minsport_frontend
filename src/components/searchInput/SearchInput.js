import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import Input from 'components/input/Input'

import 'components/searchInput/SearchInput.scss'

const SearchInput = ({ className, placeholder, onChange, value }) => (
  <div
    className={classnames([
      'search-input',
      'control',
      'has-icons-left',
      className,
    ])}
  >
    <Input
      className="input is-small"
      type="text"
      placeholder={placeholder}
      value={value}
      onChange={onChange}
    />
    <span className="icon is-small is-left">
      <i className="ic-search" />
    </span>
  </div>
)

SearchInput.defaultProps = {
  className: '',
  placeholder: 'Поиск',
}

SearchInput.propTypes = {
  className: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
}

export default SearchInput
