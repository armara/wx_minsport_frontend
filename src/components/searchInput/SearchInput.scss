@import "styles/mixins";

.search-input {
  .input {
    @include input-text();
    box-shadow: none;
    border-radius: 3px;
    border: solid 1px $smoke-white;

    & + .icon {
      height: 100%;
    }

    & + span {
      margin-left: 5px;
    }
  }
}