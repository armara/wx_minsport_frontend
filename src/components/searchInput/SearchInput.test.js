import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import SearchInput from 'components/searchInput/SearchInput'

configure({ adapter: new Adapter() })

describe('<SearchInput />', () => {
  it('renders correctly default props', () => {
    const tree = renderer
      .create(<SearchInput onChange={() => {}} value="test" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
