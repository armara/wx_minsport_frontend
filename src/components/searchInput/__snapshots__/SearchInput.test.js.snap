// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> renders correctly default props 1`] = `
<div
  className="search-input control has-icons-left"
>
  <input
    className="input is-small base-input input"
    disabled={false}
    onChange={[Function]}
    placeholder="Поиск"
    type="text"
    value="test"
  />
  <span
    className="icon is-small is-left"
  >
    <i
      className="ic-search"
    />
  </span>
</div>
`;
